### PR TITLE
Release run loop when resolving proxy on macOS.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

**Comment by nmoinvaz** [comment]
@sergio-nsk do you think any of this PR is worth keeping?

**Comment by sergio-nsk** [comment]
> @sergio-nsk do you think any of this PR is worth keeping?

`if (!run_loop)` should be kept. Not sure about `CFRelease (run_loop)`. Need take a look at Apple sources.

**Comment by sergio-nsk** [comment]
As for [WebKit](https://github.com/apple-oss-distributions/WebCore/blob/d8528fda3e9828ad9a6d3d7d3bd899a671abe7b1/platform/network/cf/ProxyServerCFNet.cpp#L84), `CFRelease(run_loop)` must be called after `CFRunLoopSourceInvalidate(run_loop)`.

File: resolver_mac.c [hunk 1]
```diff
@@ -148,10 +148,16 @@ bool proxy_resolver_mac_get_proxies_for_url(void *ctx, const char *url) {
...
+        if (!run_loop) {
+            proxy_resolver->error = ELOOP;
+            LOG_ERROR("Failed to execute pac url (%" PRId32 ")\n", proxy_resolver->error);
+            goto mac_done;
+        }
 
         CFRunLoopAddSource(CFRunLoopGetCurrent(), run_loop, PROXY_RESOLVER_RUN_LOOP);
         CFRunLoopRunInMode(PROXY_RESOLVER_RUN_LOOP, PROXY_RESOLVER_TIMEOUT_SEC, false);
         CFRunLoopRemoveSource(CFRunLoopGetCurrent(), run_loop, PROXY_RESOLVER_RUN_LOOP);
+        CFRelease(run_loop);
```
Comment (sergio-nsk) [inline resolver_mac.c:160]: As for [WebKit](https://github.com/apple-oss-distributions/WebCore/blob/d8528fda3e9828ad9a6d3d7d3bd899a671abe7b1/platform/network/cf/ProxyServerCFNet.cpp#L84), `CFRelease(run_loop)` must be called after `CFRunLoopSourceInvalidate(run_loop)`.

**Review by sergio-nsk — CHANGES_REQUESTED** [review]

File: resolver_mac.c (hunk 1 above)
Comment (nmoinvaz) [reply resolver_mac.c:161]: Ok, I have added `CFRunLoopSourceInvalidate`.

**Review by sergio-nsk — APPROVED** [review]

**Pull request merged**